platform/ios/src/MGLMapView.h

```diff
@@ -1206,9 +1206,6 @@ MGL_EXPORT IB_DESIGNABLE
 /**
  Selects an annotation and displays a callout view for it.
 
- If the given annotation is not visible within the current viewport, this
- method has no effect.
-
  @param annotation The annotation object to select.
  @param animated If `YES`, the callout view is animated into position.
```

Comment:
This needs to be updated to say that the viewport moves to show the annotation.

Reply:
This seems like a good place to talk about the `animated` parameter generally.

As implemented, the `animated` param controls whether or not the viewport is moved to show the annotation, not whether or not the viewport is moved _with an animation_. This seems potentially confusing.

I think the `animated` param should control _how_ the viewport moves to show the annotation, not whether or not the viewport moves at all. For most cases, this would be what the developer expects: selecting an annotation indicates that it should probably be visible.

I’m less certain about how to or if we should handle the case where a newly selected annotation shouldn’t move the viewport.

Reply:
Yes, this is confusing 😢 As currently written it should match Mapkit, which is also confusing. I don't like the use if the word `animated` in this context. 

I'm tempted to leave the functionality as is, but really try to make it clear through documentation. 

Perhaps changing `animated` to something more appropriate in the new (not yet public) method, would help here, and pave the way for clarifying (or removing) the existing method.